vmm/src/api/server.rs

```diff
@@ -32,4 +68,28 @@ impl<'a> ApiServer<'a> {
             num_threads,
         }
     }
+
+    pub fn serve(&self) -> Result<()> {
+        let listener = UnixListener::bind(self.path).map_err(Error::Bind)?;
+        let pool = ThreadPool::new(self.num_threads);
+
+        for stream in listener.incoming() {
+            match stream {
+                Ok(s) => {
+                    pool.execute(move || {
```

Comment:
I'm not sure having multi-threaded handling of the HTTP requests is needed here. Are we really going to see a big performance difference?
I'm asking because we're pulling a new crate for this purpose.